website/docs/r/resource_deployment_script_azure_cli.html.markdown

```diff
@@ -0,0 +1,148 @@
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+resource "azurerm_resource_deployment_script_azure_cli" "example" {
+  name                = "example-rdsac"
+  resource_group_name = azurerm_resource_group.example.name
+  location            = "West Europe"
+  version             = "2.40.0"
+  retention_interval  = "P1D"
+  arguments           = "'foo' 'bar'"
+  cleanup_preference  = "OnSuccess"
+  force_update_tag    = "1"
+  timeout             = "PT30M"
+
+  script_content = <<EOF
+            echo "{\"name\":{\"displayName\":\"$1 $2\"}}" > $AZ_SCRIPTS_OUTPUT_PATH
+  EOF
+
+  identity {
+    type = "UserAssigned"
+    identity_ids = [
+      azurerm_user_assigned_identity.example.id
+    ]
+  }
+
+  tags = {
+    key = "value"
+  }
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name which should be used for this Resource Deployment Script. The name length must be from 1 to 260 characters. The name can only contain alphanumeric, underscore, parentheses, hyphen and period, and it cannot end with a period. Changing this forces a new Resource Deployment Script to be created.
+
+* `resource_group_name` - (Required) Specifies the name of the Resource Group where the Resource Deployment Script should exist. Changing this forces a new Resource Deployment Script to be created.
+
+* `location` - (Required) Specifies the Azure Region where the Resource Deployment Script should exist. Changing this forces a new Resource Deployment Script to be created.
+
+* `version` - (Required) Azure CLI module version to be used. The supported versions are `2.0.77`, `2.0.78`, `2.0.79`, `2.0.80`, `2.0.81`, `2.1.0`, `2.10.0`, `2.10.1`, `2.11.0`, `2.11.1`, `2.12.0`, `2.12.1`, `2.13.0`, `2.14.0`, `2.14.1`, `2.14.2`, `2.15.0`, `2.15.1`, `2.16.0`, `2.17.0`, `2.17.1`, `2.18.0`, `2.19.0`, `2.19.1`, `2.2.0`, `2.20.0`, `2.21.0`, `2.22.0`, `2.22.1`, `2.23.0`, `2.24.0`, `2.24.1`, `2.24.2`, `2.25.0`, `2.26.0`, `2.26.1`, `2.27.0`, `2.27.1`, `2.27.2`, `2.28.0`, `2.29.0`, `2.29.1`, `2.29.2`, `2.3.0`, `2.3.1`, `2.30.0`, `2.31.0`, `2.32.0`, `2.33.0`, `2.33.1`, `2.34.0`, `2.34.1`, `2.35.0`, `2.36.0`, `2.37.0`, `2.38.0`, `2.39.0`, `2.4.0`, `2.40.0`, `2.41.0`, `2.5.0`, `2.5.1`, `2.6.0`, `2.7.0`, `2.8.0`, `2.9.0`, `2.9.1`. Changing this forces a new Resource Deployment Script to be created.
+
+* `retention_interval` - (Required) Interval for which the service retains the script resource after it reaches a terminal state. Resource will be deleted when this duration expires. The time duration should be between `1` hour and `26` hours (inclusive) and should be specified in ISO 8601 format. Changing this forces a new Resource Deployment Script to be created.
+
+* `arguments` - (Optional) Command line arguments to pass to the script. Arguments are separated by spaces. ex: `-Location 'West US 2'`. Changing this forces a new Resource Deployment Script to be created.
```

Comment:
this should be an array of strings given "Arguments are separated by spaces"

Reply:
Since the `arguments` is defined as `string` in REST API, shall we use the same type in the Provider?
The reason is: if we define `arguments` as a string list, we need to join the elements with space (" ") in `Create` function and split the string in `Read` function. Some argument strings may cause unexpected result when splitting. For example, if the user set arguments =  [`"-name", "\"John Dole\""`], it will be joined as `"-name \"John Dole\""`. If we use space to split the string, we will get the list ["-name", "\"John", "Dole\""] which is different from the original one. Though we could use `strings.Fields` function to solve this issue, there may exist other specific argument strings that cannot be correctly split by `strings.Fields`.

Reply:
i'm not sure how `-Location 'West US 2'` is any different from `["-Location", "West US 2"]` ?

Reply:
> i'm not sure how `-Location 'West US 2'` is any different from `["-Location", "West US 2"]` ?

Since `arguments` is defined as `string` in Go SDK, if we define this property as a string list in the schema, we need to join `["-Location", "'West US 2'"]` into `-Location 'West US 2'` in `create` function and convert `-Location 'West US 2'` to `["-Location", "'West US 2'"]` in `read` function. In `read` function, if we split `-Location 'West US 2'` with a space, we will get `["-Location", "'West", "US", "2'"]`, which is different from `["-Location", "'West US 2'"]`

Reply:
Ohhh, I do see where you are coming from here but I think the way it is written is causing a bit of confusion. What do you think about rewriting that attribute to be `command_line` instead of `arguments`. Then we can remove 
```
Arguments are separated by spaces. ex: `-Location 'West US 2'`
```
since `command_line` with the description `Command line arguments to pass to the script.` is more than clear on how arguments should be passed (i.e. just pass in arguments here the same as you would the command line)